app/javascript/crayons/MarkdownToolbar/__tests__/markdownSyntaxFormatters.test.js

```diff
@@ -1516,6 +1516,239 @@ describe('markdownSyntaxFormatters', () => {
         expect(newCursorEnd).toEqual(4);
       });
     });
+
+    describe('embed', () => {
+      it('inserts embed syntax and highlights url, when no selection is given', () => {
+        const textAreaValue = 'one two three';
+        const expectedNewTextAreaValue = 'one two {% embed https://... %}three';
+
+        const {
+          newCursorStart,
+          newCursorEnd,
+          editSelectionStart,
+          editSelectionEnd,
+          replaceSelectionWith,
+        } = coreSyntaxFormatters['embed'].getFormatting({
+          value: textAreaValue,
+          selectionStart: 8,
+          selectionEnd: 8,
+        });
+
+        const editedString = getNewTextAreaValueWithEdits({
+          textAreaValue,
+          editSelectionStart,
+          editSelectionEnd,
+          replaceSelectionWith,
+        });
+
+        expect(editedString).toEqual(expectedNewTextAreaValue);
+        expect(editedString.substring(newCursorStart, newCursorEnd)).toEqual(
+          'https://...',
+        );
+      });
+
+      it('inserts embed syntax and highlights text, when selected text does not begin with http:// or https://', () => {
```

Comment:
Actually - ignore this - I see the different handling of the URLs, and these tests make perfect sense!

Reply:
Actually I did thought so too that we don't need URL and non-URL tests for this feature but added it anyways because there can never be enough tests.
Nevertheless, should I update these two tests to combine into one test?

Reply:
I don't feel strongly about it, so happy to defer to whatever you think is best. I think the only downside to having the extra test is that it gives the impression to the next developer that the non-URL vs URL cases should be handled differently 🤷‍♀️

Reply:
In that case I will revert it. Thanks a lot.

Reply:
I have updated all the test cases as per your comments and @amywtlin 's comments.